portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -47,15 +49,23 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     // Get prepared instance of the query
     PreparedQuery results = datastore.prepare(query);
 
+
+
     // Iterate over results
     List<Comment> comments = new ArrayList<>();
+    int count = 0;
     for (Entity entity : results.asIterable()) {
+      if (count >= maxComments) {
+          break;
+      }
+
```

Comment:
This might be a bit easier to read if you make the for loop over the count index. i.e.

for (int count = 0; count < maxComments; count++) {
  Entity entity = results.next()
  ...
}

Reply:
I pushed commtis to all the other suggestions but I had a lot of trouble finding a way to iterate over a preparedQuery as a list. I tried using the asList method but that did not seem to work. I also tried the next method but that was not compatible with the type of results. I was stuck on trying to make that adjustment so I ended up not changing it yet. Would you have any suggestions on how to approach that?

Reply:
you should be able to do the following:
```
Iterator<Entity> iter = results.AsIterator()
for(...) {
  Entity entity = iter.next()
  ...
}
```

Reply:
See https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/PreparedQuery.html#asIterator--

Reply:
Thank you! I got it to work now. Just committed the changes.